libraries/rush-lib/src/logic/Autoinstaller.ts

```diff
@@ -181,6 +182,19 @@ export class Autoinstaller {
       oldFileContents = FileSystem.readFile(this.shrinkwrapFilePath, { convertLineEndings: NewlineKind.Lf });
       this._logIfConsoleOutputIsNotRestricted('Deleting ' + this.shrinkwrapFilePath);
       FileSystem.deleteFile(this.shrinkwrapFilePath);
+      if (this._rushConfiguration.packageManager === 'pnpm') {
+        // Workaround for https://github.com/pnpm/pnpm/issues/1890
+        //
+        // When "rush update-autoinstaller" is run, Rush deletes "common/autoinstallers/my-task/pnpm-lock.yaml"
+        // so that a new lockfile will be generated. However "pnpm install" by design will try to recover
+        // "pnpm-lock.yaml" from "my-task/node_modules/.pnpm/lock.yaml", which may prevent a full upgrade.
+        // Deleting both files ensures that a new lockfile will always be generated.
```

Comment:
Is there a CLI parameter we can pass to pnpm to do this instead? Deleting files feels a little jank.

Reply:
This is simply the same algorithm that `rush update` uses.  

Why does Rush do it that way? If I remember, the approach originated with the NPM package manager, whose CLI and behavior for updating lockfiles was inconsistent across versions and maybe had some bugs. Deleting the lockfile was a reliable way to ensure that the old installation plan was completely discarded. This same approach was copied when moving to PNPM, but then we needed this workaround because of the backup file.

I believe the idiomatic way with PNPM is just `pnpm update` instead of `pnpm install`. However if we want to switch to this CLI, it is a nontrivial change to Rush that needs to be carefully tested. I wouldn't do that in this PR.

Whatever installation method we use, it's probably best to keep it consistent between autoinstallers and regular `rush install`.